tests/end_to_end.rs

```diff
@@ -1071,3 +1072,172 @@ async fn test_vacuum_command() {
+    assert_partition_ids(&context, 5, vec![1, 2, 3, 4]).await;
+
+    //
+    // Execute DELETE affecting partitions 2, 3 and creating table_version 6
+    //
+    let plan = context
+        .plan_query("DELETE FROM test_table WHERE some_value > 46")
+        .await
+        .unwrap();
+    context.collect(plan).await.unwrap();
+
+    assert_partition_ids(&context, 6, vec![1, 4, 5]).await;
+
+    let partitions = context
+        .partition_catalog
+        .load_table_partitions(6 as TableVersionId)
+        .await
+        .unwrap();
+
+    // Assert result of the new partition with id 5
+    let results = scan_partitions(&context, Some(vec![4]), partitions[2].clone()).await;
+    let expected = vec![
+        "+------------+",
+        "| some_value |",
+        "+------------+",
+        "| 45         |",
+        "| 46         |",
+        "| 46         |",
+        "+------------+",
+    ];
+    assert_batches_eq!(expected, &results);
+
+    // Verify results
+    let plan = context
+        .plan_query("SELECT some_value FROM test_table")
+        .await
+        .unwrap();
+    let results = context.collect(plan).await.unwrap();
+
+    // TODO: the partition order is swapped here
+    let expected = vec![
+        "+------------+",
+        "| some_value |",
+        "+------------+",
+        "| 42         |",
+        "| 43         |",
+        "| 44         |",
+        "| 42         |",
+        "| 41         |",
+        "| 40         |",
+        "| 45         |",
+        "| 46         |",
+        "| 46         |",
+        "+------------+",
+    ];
+    assert_batches_eq!(expected, &results);
+
+    //
+    // Execute DELETE with multiple conditions, removing entire partition 4, and trimming partitions 1 and 5
+    //
+    let plan = context
+        .plan_query("DELETE FROM test_table WHERE some_value < 43 OR some_value > 45")
+        .await
+        .unwrap();
+    context.collect(plan).await.unwrap();
+
+    assert_partition_ids(&context, 7, vec![6]).await;
+
+    // Verify results
+    let plan = context
+        .plan_query("SELECT some_value FROM test_table")
+        .await
+        .unwrap();
+    let results = context.collect(plan).await.unwrap();
+
+    let expected = vec![
+        "+------------+",
+        "| some_value |",
+        "+------------+",
+        "| 43         |",
+        "| 44         |",
+        "| 45         |",
+        "+------------+",
+    ];
+    assert_batches_eq!(expected, &results);
+
+    //
+    // Execute blank DELETE, without qualifiers
+    //
+    let plan = context.plan_query("DELETE FROM test_table").await.unwrap();
+    context.collect(plan).await.unwrap();
+
+    assert_partition_ids(&context, 8, vec![]).await;
+
+    // Verify results
+    let plan = context
+        .plan_query("SELECT some_value FROM test_table")
+        .await
+        .unwrap();
+    let results = context.collect(plan).await.unwrap();
+
+    assert!(results.is_empty());
```

Comment:
Another test suggestion: a `DELETE` that doesn't do anything (either because all partitions get pruned out or because we process partitions but nothing in there matches the qualifiers)

Reply:
I did add such a test; note that this is not a true no-op, as a new table version (identical to the prior one) gets created.